File: src/main/java/com/wesker926/leetcode/algorithms/p0520/s1/Solution.java
package com.wesker926.leetcode.algorithms.p0520.s1;

/**
 * @author wesker.gh
 * @date 2021/11/13
 * @description 遍历
 */
public class Solution {
    public boolean detectCapitalUse(String word) {
        boolean first = Character.isUpperCase(word.charAt(0));
        int up = 0;
        for (char c : word.toCharArray()) {
            if (Character.isUpperCase(c)) {
                up++;
            }
        }
        return (first && up == word.length()) || (first && up == 1) || (!first && up == 0);
    }
}

File: src/main/java/com/wesker926/leetcode/algorithms/p0520/s1/solution.go
// Package s1
// @author: wesker.gh
// @date: 2021/11/13
// @description: 遍历
package s1

import "unicode"

func detectCapitalUse(word string) bool {
	first, up := unicode.IsUpper(rune(word[0])), 0
	for _, c := range word {
		if unicode.IsUpper(c) {
			up++
		}
	}
	return (first && up == len(word)) || (first && up == 1) || (!first && up == 0)
}
